src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -1062,8 +1062,8 @@ impl WithOptionName for KafkaSinkConfigOptionName {
             | KafkaSinkConfigOptionName::LegacyIds
             | KafkaSinkConfigOptionName::TopicConfig
             | KafkaSinkConfigOptionName::TopicPartitionCount
-            | KafkaSinkConfigOptionName::TopicReplicationFactor => false,
-            KafkaSinkConfigOptionName::PartitionBy => true,
+            | KafkaSinkConfigOptionName::TopicReplicationFactor
+            | KafkaSinkConfigOptionName::PartitionBy => false,
```

Comment:
We might want to keep these values as redacted since it includes field names? Up to you though!